profiler/src/main/java/eu/clarin/switchboard/profiler/DefaultProfiler.java

```diff
@@ -0,0 +1,100 @@
+        jsonProfiler = new JsonProfiler();
+        textProfiler = new TextProfiler();
+        textExtractor = new TikaTextExtractor(tikaConfig);
+        languageDetector = new OptimaizeLanguageDetector();
+    }
+
+    public List<Profile> profile(File file) throws IOException, ProfilingException {
+        List<Profile> profiles;
+        Profile firstProfile;
+        {
+            profiles = tikaProfiler.profile(file);
+            LOGGER.debug("file " + file.getName() + "; tika returned: " + profiles);
+            assert profiles.size() == 1;
+            firstProfile = profiles.get(0);
+        }
+
+        if (firstProfile.isMediaType(MediaType.APPLICATION_XML)) {
+            List<Profile> xmlProfiles = xmlProfiler.profile(file);
+            LOGGER.debug("file " + file.getName() + "; xmlprof returned: " + xmlProfiles);
+            if (xmlProfiles != null && !xmlProfiles.isEmpty()) {
+                profiles = xmlProfiles;
+                firstProfile = profiles.get(0);
+            }
+        } else if (firstProfile.isMediaType(MediaType.APPLICATION_JSON)) {
+            List<Profile> jsonProfiles = jsonProfiler.profile(file);
+            LOGGER.debug("file " + file.getName() + "; jsonprof returned: " + jsonProfiles);
+            if (jsonProfiles != null && !jsonProfiles.isEmpty()) {
+                assert jsonProfiles.size() == 1;
+                profiles = jsonProfiles;
+                firstProfile = profiles.get(0);
+            }
+        } else if (firstProfile.isMediaType(MediaType.TEXT_PLAIN)) {
+            List<Profile> textProfiles = textProfiler.profile(file);
+            LOGGER.debug("file " + file.getName() + "; textprof returned: " + textProfiles);
+            if (textProfiles != null && !textProfiles.isEmpty()) {
+                profiles = textProfiles;
+                firstProfile = profiles.get(0);
+            }
+        }
+
+        if (firstProfile.getMediaType() == null) {
+            firstProfile = Profile.builder(firstProfile).mediaType(MediaType.APPLICATION_OCTET_STREAM).build();
+            profiles = new ArrayList<>(profiles);
+            profiles.set(0, firstProfile);
+        } else if (firstProfile.getLanguage() == null) {
+            try {
+                String text = textExtractor.getText(file, firstProfile.getMediaType());
+                String language = languageDetector.detect(text);
+                LOGGER.debug("file " + file.getName() + "; detected language: " + language);
+                if (language != null) {
+                    firstProfile = Profile.builder(firstProfile).language(language).build();
+                    profiles = new ArrayList<>(profiles);
+                    profiles.set(0, firstProfile);
+                }
+            } catch (IOException | TikaException xc) {
+                LOGGER.info("Cannot convert media to text for detecting the language: " + xc.getMessage());
+            }
+        }
+
+        return profiles;
+    }
```

Comment:
I really need some javadoc to understand what's happening here and why. I also see a lot of (near) repetition, perhaps could be refactored to reduce this?